File: apps/backend/src/EnterpriseAgentOs.Domain/features/Billing/IUserBillingService.cs
namespace EnterpriseAgentOs.Domain.Features.Billing;

public interface IUserBillingService
{
    Task<UserSubscription> GetSubscriptionAsync(Guid userId, CancellationToken ct = default);
    Task<(long Remaining, bool OverBudget)> CheckCreditBudgetAsync(Guid userId, CancellationToken ct = default);
    Task<string> CreateCheckoutSessionAsync(Guid userId, string email, string plan, string billingCycle, CancellationToken ct = default);
    Task<string> CreatePortalSessionAsync(Guid userId, string email, CancellationToken ct = default);
    Task EnableOverageAsync(Guid userId, string email, bool enabled, CancellationToken ct = default);
    Task<IReadOnlyList<InvoicePayload>> ListInvoicesAsync(
        Guid userId,
        CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/management/IUserBillingService.java
package com.enterpriseagentos.backendjava.domain.features.management;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.enterpriseagentos.backendjava.domain.common.primitives.*;
import com.enterpriseagentos.backendjava.domain.common.services.*;
import com.enterpriseagentos.backendjava.domain.common.valueobjects.*;
import com.enterpriseagentos.backendjava.domain.events.*;
import com.enterpriseagentos.backendjava.domain.features.agents.*;
import com.enterpriseagentos.backendjava.domain.features.analytics.*;
import com.enterpriseagentos.backendjava.domain.features.management.*;
import com.enterpriseagentos.backendjava.domain.features.mcp.*;

public interface IUserBillingService {
    CompletableFuture<UserSubscription> getSubscriptionAsync(UUID userId);
    CompletableFuture<CreditBudgetResult> checkCreditBudgetAsync(UUID userId);
    CompletableFuture<String> createCheckoutSessionAsync(UUID userId, String email, String plan, String billingCycle);
    CompletableFuture<String> createPortalSessionAsync(UUID userId, String email);
    CompletableFuture<Void> enableOverageAsync(UUID userId, String email, boolean enabled);
}
